distdir_deps.bzl

```diff
@@ -35,3 +35,66 @@ DIST_DEPS = {
         ],
     },
 }
+
+def _gen_workspace_stanza_impl(ctx):
+    if ctx.attr.template and (ctx.attr.preamble or ctx.attr.postamble):
+        fail("Can not use template with either preamble or postamble")
+
+    repo_clause = """
+maybe(
+    http_archive,
+    "{repo}",
+    sha256 = "{sha256}",
+    strip_prefix = {strip_prefix},
+    urls = {urls},
+)
+"""
+    repo_stanzas = {}
+    for repo in ctx.attr.repos:
+        info = DIST_DEPS[repo]
```

Comment:
Throw an error if repo is not in DIST_DEPS?

Reply:
It will throw an an error with a call stack. That is the most useful thing because it pinpoints where I incorrectly used the name.